src/lib.rs

```diff
@@ -345,12 +346,27 @@ pub mod pallet {
 
 			let current_initialized_rewards = InitializedRewardAmount::<T>::get();
 
-			// This ensures that the rewards match the pot
+			let reward_difference = Self::pot().saturating_sub(current_initialized_rewards);
+
+			// Ensure the difference is not bigger than the total number of contributors
 			ensure!(
-				current_initialized_rewards == Self::pot(),
+				reward_difference < TotalContributors::<T>::get().into(),
 				Error::<T>::RewardsDoNotMatchFund
 			);
 
+			if reward_difference > 0u32.into() {
+				let to_burn = T::RewardCurrency::burn(reward_difference);
+
+				// Shouldnt fail, as the fund should be enough to burn and nothing is locked
+				T::RewardCurrency::settle(
+					&PALLET_ID.into_account(),
+					to_burn,
+					WithdrawReasons::TRANSFER,
+					AllowDeath,
+				)
+				.map_err(|_| Error::<T>::ErrorBurningImbalance)?;
```

Comment:
```suggestion
				// Shouldnt fail, as the fund should be enough to burn and nothing is locked
				T::RewardCurrency::settle(
					&PALLET_ID.into_account(),
					to_burn,
					WithdrawReasons::TRANSFER,
					AllowDeath,
				)
				.expect(Shouldnt fail, as the fund should be enough to burn and nothing is locked);
```

When there is no way for it to fail you can use `expect` and avoid having to add a new error variant that will never be used.

Reply:
cool